### PR TITLE
added alchemy and provider support

### PR DESCRIPTION
**Pull request by PatrickAlphaC**
### What I did

Added support for Alchemy built-in

Related issue: #

https://github.com/eth-brownie/brownie/issues/1177

### How I did it

I added a `providers-config.yaml` to the `.brownie` dir. Now, if you want to switch from Infura to Alchemy, you set your `WEB3_ALCHEMY_PROJECT_ID` and run:

```
brownie networks set_provider alchemy
```

### How to verify it

Try it out!

1. Add a `provider` flag to your infura network of choice.

```
brownie networks modify mainnet provider=infura
```

Then, swap it to alchemy

```
brownie networks set_provider alchemy
```

### Checklist

- [ ] I have confirmed that my PR passes all linting checks
- [x] I have included test cases
- [x] I have updated the documentation
- [x] I have added an entry to the changelog

For some reason, a few tests locally keep failing for me... wondering if I can get some help cleaning up the last few.

### CODE REVIEW

**Review by iamdefinitelyahuman — APPROVED** [review]
This is super cool!

File: docs/network-management.rst [hunk 1]
```diff
@@ -206,6 +258,26 @@ To connect to Infura using Brownie, store your project ID as an environment vari
 
     $ export WEB3_INFURA_PROJECT_ID=YourProjectID
 
+Or adding `export WEB3_INFURA_PROJECT_ID=YourProjectID` to your `.env` and adding `dotenv: .env` to your `brownie-config.yaml`. 
+
+
+Using Alchemy
+*************
+
+To Infura you need to `signup for an account <https://auth.alchemyapi.io/signup>`_. Once you have signed up, login and create a new project. You will be provided with a URL that can be leveraged to access the network.
```
Comment (iamdefinitelyahuman) [inline docs/network-management.rst:267]: To infura?

Comment (PatrickAlphaC) [reply docs/network-management.rst:267]: I'll fix 😅

**Comment by PatrickAlphaC** [comment]
Updated!

**Review by iamdefinitelyahuman — APPROVED** [review]

**Comment by HetmanJones** [comment]
I'm wonder if it is possible to do via configuration file? Thanks

**Pull request merged**